src/index.ts

```diff
@@ -121,12 +125,29 @@ export class Pipeline implements Serializable {
       : {
           explicitDependencies: false,
         };
-
+    const steps = await this.toList(newOpts);
+    if (opts.mutator) {
+      for (const step of steps) {
+        if (step instanceof Step) {
+          const deps = {
+            dependencies: cloneDeep(step.dependencies),
+            effectDependencies: cloneDeep(step.effectDependencies),
+          };
+          await opts.mutator(step);
+          if (
+            !isEqual(deps, {
```

Comment:
I don't think this should use `isEqual` since it's a deep comparison. We don't want to compare deeply, we want to make sure they still refer to the same step objects as before.

Reply:
`isEqual` handles object identity so it's ok: https://github.com/lodash/lodash/blob/2da024c3b4f9947a48517639de7560457cd4ec6c/.internal/baseIsEqual.js#L19-L21